src/appleseed/renderer/modeling/environmentedf/physicalsky.h

```diff
@@ -0,0 +1,222 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2020 Joel Barmettler, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+#include "foundation/math/vector.h"
+#include "foundation/image/regularspectrum.h"
+#include "foundation/math/ray.h"
+#include "foundation/math/distance.h"
+
+#include "atmosphereshell.h"
+#include "opticaldepth.h"
+
+using namespace foundation;
+
+namespace nishita {
+
+    static const int num_wavelengths = 31;                                     // number of wavelengths per spectrum (400nm to 700nm, delta=10nm)
+    static const int n_cylinders = 1024;                                       // number of cylinders for optical depth precomputation
+
+    // Lookup table storing optical dephts into the direction of the sun.
+    static sky::opticaldepth optical_depths_table[shell::n_atmosphere_shells + 1][n_cylinders];
+
+    // Width of a single cylinder. Sum of all cylinder widths must be marginally smaller than the atmosphere radius.
+    static const float cylinder_delta = (atmosphere_radius - n_cylinders) / (n_cylinders - 1);
+
+    // Mie scattering coefficient and regular spectrum of mie coefficients.
+    const float mie_extinction_coeff = 2e-5f;
+    const RegularSpectrum31f mie_coeff_spectrum = RegularSpectrum31f(mie_extinction_coeff);
+
+    // Sun irradiance (W*m^-2*nm^-1) values at the top of the atmosphere.
+    // Source: https://www.nrel.gov/grid/solar-resource/spectra.html, Table SMART MODTRAN ETR Spectra.
+    const float sun_radiance[num_wavelengths] = {
```

Comment:
These values are already present in `src/modeling/light/sunlight.cpp`. Can we join them somehow?

Reply:
Let's discuss a good solution for joining these arrays before I touch someone elses code :)